crates/ruff_python_parser/resources/invalid/expressions/dict/recover.py

```diff
@@ -0,0 +1,27 @@
+# Test cases for dictionary expressions where the parser recovers from a syntax error.
+
+{,}
+
+{1: 2,,3: 4}
+
+{1: 2,,}
+
+# Missing comma
+{1: 2 3: 4}
+
+# No value
+{1: }
+
+# No value for double star unpacking
+{**}
+{x: y, **, a: b}
+
+# This is not a double star unpacking
+# {* *data}
+
+# Star expression not allowed here
+# TODO(dhruvmanila): Disallow star expression `b`
+{*x: y, z: a, *b: c}
+
+# TODO(dhruvmanila): Disallow star expression
+# {x: *y, z: *a}
```

Comment:
This needs to be solved holistically similar to how it was done in star pattern parsing.